batchflow/opensets/pascal.py

```diff
@@ -0,0 +1,179 @@
+""" Contains PascalVOC dataset and labels for different tasks """
+
+import os
+from os.path import dirname, basename
+from io import BytesIO
+import tarfile
+import tempfile
+import urllib
+from collections import defaultdict
+import logging
+
+import PIL
+import tqdm
+import numpy as np
+
+from . import ImagesOpenset
+from .. import DatasetIndex
+
+logger = logging.getLogger('PascalVOCdataset')
+
+class BasePascal(ImagesOpenset):
+    """ The base class for PascalVOC dataset.
+    The archive contains 17125 images. Total size 1.9GB.
+
+    Tracks of the PascalVOC challenge:
+        1. Classification
+        2. Detection
+        3. Segmentation
+        4. Action Classification Task
+        5. Boxless Action Classification
+        6. Person Layout
+
+    Notes
+    -----
+    - Each track contains only the subset of the total images with labels provided.
+    """
+    SOURCE_URL = 'http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar'
+    sets_path = 'VOCdevkit/VOC2012/ImageSets'
+    task = None
+
+    def __init__(self, *args, bar=False, preloaded=None, train_test=True, **kwargs):
```

Comment:
Since the dataset is very large allow for storing it on a disk and working with files as well.